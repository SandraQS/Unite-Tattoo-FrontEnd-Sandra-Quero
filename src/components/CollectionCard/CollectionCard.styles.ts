import { StyleSheet } from "react-native";
import { colors } from "../../styles/uniteTatto.styles";

export default StyleSheet.create({
  collectionCard: {
    backgroundColor: colors.lightBrown,
    borderRadius: 20,
    flex: 1,
    flexDirection: "column",
    margin: 5,
    padding: 10,
  },

  imageCollection: {
    borderRadius: 20,
    flex: 5,
  },

  imageContainer: {
    alignContent: "center",
    borderRadius: 20,
    flex: 5,
    height: 200,
    justifyContent: "center",
  },

  mainContainerCard: {
    height: 300,
    width: 170,
  },

  styleTitle: {
    color: colors.darkBrown,
    fontSize: 20,
    fontWeight: "800",
  },

  textContainer: {
    flex: 1,
    justifyContent: "flex-end",
  },
});
