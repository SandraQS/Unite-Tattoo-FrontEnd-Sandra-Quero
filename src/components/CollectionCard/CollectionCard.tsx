import React, { useState } from "react";
import { Image, Text, TouchableOpacity, View } from "react-native";
import { EditDelete } from "../EditDelete/EditDelete";

import style from "./CollectionCard.styles";

const CollectionCard = ({ collection }: any) => {
  const [longPress, setLongPress] = useState(false);
  const showButtons = () => {
    setLongPress(!longPress);
  };

  const onPressCollection = () => {
    if (longPress) {
      setLongPress(false);
    } else {
      //ir a trabajos
    }
  };

  return (
    <TouchableOpacity
      onPress={onPressCollection}
      onLongPress={showButtons}
      activeOpacity={0.6}
    >
      <View style={style.mainContainerCard}>
        <View style={style.collectionCard}>
          <View style={style.imageContainer}>
            <Image
              source={{ uri: collection.image }}
              style={style.imageCollection}
            />
          </View>

          <View style={style.textContainer}>
            <Text style={style.styleTitle}>{collection.tattooStyles}</Text>
          </View>
          <View style={style.buttons}>{longPress && <EditDelete />}</View>
        </View>
      </View>
    </TouchableOpacity>
  );
};

export default CollectionCard;
