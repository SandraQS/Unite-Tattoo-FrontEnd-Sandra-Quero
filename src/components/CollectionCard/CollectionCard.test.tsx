import { render } from "@testing-library/react-native";
import React from "react";
import CollectionCard from "./CollectionCard";

describe("Given a CollectionCard component", () => {
  describe("When it is rendered", () => {
    test("Then it should show match the last snapshot", () => {
      const collection = {
        image:
          "https://storage.googleapis.com/unite-tattoo.appspot.com/Acuarela-1638104950642-.png",
        tattooStyles: "Acuarela",
        id: "2",
      };
      const functionGoWorks = jest.fn();
      const functionGoEdit = jest.fn();

      const screen = render(
        <CollectionCard
          functionGoWorks={functionGoWorks}
          collection={collection}
          functionGoEdit={functionGoEdit}
        />
      );

      expect(screen).toMatchSnapshot();
    });
  });
});
