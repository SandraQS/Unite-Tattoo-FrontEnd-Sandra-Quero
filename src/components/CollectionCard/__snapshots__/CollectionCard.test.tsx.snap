// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given a CollectionCard component When it is rendered Then it should show match the last snapshot 1`] = `
<View
  accessible={true}
  collapsable={false}
  focusable={true}
  nativeID="animatedComponent"
  onClick={[Function]}
  onResponderGrant={[Function]}
  onResponderMove={[Function]}
  onResponderRelease={[Function]}
  onResponderTerminate={[Function]}
  onResponderTerminationRequest={[Function]}
  onStartShouldSetResponder={[Function]}
  style={
    Object {
      "opacity": 1,
    }
  }
>
  <View
    style={
      Object {
        "height": 300,
        "width": 170,
      }
    }
  >
    <View
      style={
        Object {
          "backgroundColor": "#DFD8D2",
          "borderRadius": 20,
          "flex": 1,
          "flexDirection": "column",
          "margin": 5,
          "padding": 10,
          "position": "relative",
        }
      }
    >
      <View
        style={
          Object {
            "alignContent": "center",
            "borderRadius": 20,
            "flex": 5,
            "height": 200,
            "justifyContent": "center",
          }
        }
      >
        <Image
          source={
            Object {
              "uri": "https://storage.googleapis.com/unite-tattoo.appspot.com/Acuarela-1638104950642-.png",
            }
          }
          style={
            Object {
              "borderRadius": 20,
              "flex": 5,
            }
          }
        />
      </View>
      <View
        style={
          Object {
            "flex": 1,
            "justifyContent": "flex-end",
          }
        }
      >
        <Text
          style={
            Object {
              "color": "#7C6B53",
              "fontSize": 20,
              "fontWeight": "800",
            }
          }
        >
          Acuarela
        </Text>
      </View>
      <View
        style={
          Object {
            "position": "absolute",
            "right": 0,
            "top": 0,
          }
        }
      >
        <View
          style={
            Object {
              "alignItems": "center",
            }
          }
        >
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#A19178",
                "borderBottomColor": "#DFD8D2",
                "borderBottomWidth": 2,
                "borderRadius": 5,
                "opacity": 1,
                "width": 80,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#DFD8D2",
                  "fontSize": 18,
                  "padding": 5,
                  "textAlign": "center",
                }
              }
            >
              Editar
            </Text>
          </View>
          <View
            accessible={true}
            collapsable={false}
            focusable={true}
            nativeID="animatedComponent"
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "backgroundColor": "#A19178",
                "borderBottomColor": "#DFD8D2",
                "borderBottomWidth": 2,
                "borderRadius": 5,
                "opacity": 1,
                "width": 80,
              }
            }
          >
            <Text
              style={
                Object {
                  "color": "#DFD8D2",
                  "fontSize": 18,
                  "padding": 5,
                  "textAlign": "center",
                }
              }
            >
              Eliminar
            </Text>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
